cmd/object-handlers.go

```diff
@@ -1757,8 +1757,10 @@ func (api objectAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Req
 		md5hex = "" // Do not try to verify the content.
 		sha256hex = ""
 	}
-
-	hashReader, err := hash.NewReader(reader, size, md5hex, sha256hex, actualSize)
+	if crypto.S3KMS.IsNotContentMD5Requested(r.Header) {
```

Comment:
Another thing I realized is that we can do this for SSE-C requests as well so you can do 

```
if crypto.S3KMS.NoMD5(r.Header) || crypto.SSEC.NoMD5(r.Header) {
     ctx = context.WithValue(ctx, etag.ContextEtagAsUUID, mustGetUUID())
}
```